esphome/components/pcm5122/pcm_gpio.cpp

```diff
@@ -0,0 +1,39 @@
+#include "pcm_gpio.h"
+#include "esphome/core/log.h"
+
+namespace esphome {
+namespace pcm5122 {
+
+static const char *TAG = "PCM-GPIOs";
+
+void PCMGPIOPin::setup(){
+    // set input/output mode for pin in register 0x08
+    uint8_t curr = this->parent_->reg(0x08).get();
+    if( this->flags_ & gpio::FLAG_INPUT ){
+        this->parent_->reg(0x08) = curr & ~(1 << (this->pin_ - 1));
+    } else if( this->flags_ & gpio::FLAG_OUTPUT ){
+        this->parent_->reg(0x08) = curr | (1 << (this->pin_ - 1));
+        // set pin to be used as GPIO
+        this->parent_->reg(0x50 + (this->pin_ - 1)) = 0x02;  
+    }
+
+}
+
+void PCMGPIOPin::digital_write(bool value) {
+    uint8_t curr = this->parent_->reg(0x56).get();
+    if( value ){
+        this->parent_->reg(0x56) = curr | (1 << (this->pin_ - 1));
+    } else {
+        this->parent_->reg(0x56) = curr & ~(1 << (this->pin_ - 1));
+    }
+}
+
+bool PCMGPIOPin::digital_read() {
+    uint8_t reg_val = this->parent_->reg(0x77).get();
+    //ESP_LOGD(TAG, "Current value for reg 0x77: %d", reg_val);
+    return !!(reg_val & (1 << this->pin_)) != this->inverted_;
```

Comment:
This looks to be off by one. 
Likeley caused by the datasheet being a bit oddly written for Table 114. Page 0 / Register 119

From testing:
GPIN2=GPIO3
GPIN3=GPIO4
GPIN4=GPIO5
GPIN5=GPIO6

I believe the fix is 
```return !!(reg_val & (1 << (this->pin_ - 1))) != this->inverted_;```

Reply:
ahhh you already tested that as well, perfect!! I will commit the change then.